src/matches.rs

```diff
@@ -427,12 +427,10 @@ fn rewrite_match_body(
                 };
                 let semicolon = if context.config.version() == Version::One {
                     ""
+                } else if semicolon_for_expr(context, body) {
```

Comment:
warning: this `else { if .. }` block can be collapsed
   --> src\matches.rs:430:24